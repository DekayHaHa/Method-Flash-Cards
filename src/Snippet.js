import React, { Component } from 'react';
import Results from './Results'

export default class Snippet extends Component {
	constructor(props) {
		super(props);
		this.state = {
			prompt: this.props.question,
			results: false,
			answer: false,
			close: this.props.close
		}
	}
	checkAnswer = (e) => {
		e.preventDefault()
		let answer = false
		let val = this.anw.value
		if (val === this.state.prompt.solution){
			console.log(val)
			answer = true
		}
		this.setState({answer: answer, results: true})
	}
	render() {
		let displayResults = this.state.results ? <Results {...this.state}/> : <h3>PRESS ENTER TO SUBMIT</h3>
		const card = this.props.question
		return (
			<section>
				<form onSubmit={this.checkAnswer}>
					<p>{card.snippet[0]}</p>
					<p>{card.snippet[1]}
						<label htmlFor='answer'>METHOD</label>
						<input id='answer' ref={anw => this.anw = anw} type='text' />{card.snippet[2]}</p>
				<p>{card.snippet[3]}</p>
				{displayResults}
				</form>
			</section>
		);
	}
}