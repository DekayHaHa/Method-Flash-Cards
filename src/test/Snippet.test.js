import React from 'react';
import ReactDOM from 'react-dom';
import Snippet from '../Snippet';
import { mount } from 'enzyme';

const closeMock = jest.fn();
const keepScoreMock = jest.fn();
const mockPrompt = {
	"snippet": ["let arr = [1, 2, 3, 4, 5]", "let results = arr.", "(val => val > 0)", "console.log(results) // logs true"],
	"category": "Array Prototype Method",
	"solution": "every",
	"mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every",
	"uid": "card-1"
}

describe('Snippet', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(
			<Snippet close={closeMock} question={mockPrompt} keepScore={keepScoreMock}/>
		);
	});
	it('should match snapshop', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should have default state', () => {
		expect(wrapper.state()).toEqual({
			prompt: mockPrompt,
			results: false,
			answer: false,
			close: closeMock
		});
	});
	it('should checkAnswer on submit', () => {
		wrapper.find('form').simulate('submit', {
			preventDefault: () => {}
		})
		expect(keepScoreMock).toBeCalled()
	})
	it('should set state to display results', () => {
		wrapper.find('form').simulate('submit', {
			preventDefault: () => { }
		})
		expect(wrapper.state('results')).toEqual(true)
	})
})