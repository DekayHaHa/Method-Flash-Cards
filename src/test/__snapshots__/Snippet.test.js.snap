// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet should match snapshop 1`] = `
<Snippet
  close={[MockFunction]}
  keepScore={[MockFunction]}
  question={
    Object {
      "category": "Array Prototype Method",
      "mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every",
      "snippet": Array [
        "let arr = [1, 2, 3, 4, 5]",
        "let results = arr.",
        "(val => val > 0)",
        "console.log(results) // logs true",
      ],
      "solution": "every",
      "uid": "card-1",
    }
  }
>
  <section>
    <form
      onSubmit={[Function]}
    >
      <p>
        let arr = [1, 2, 3, 4, 5]
      </p>
      <p>
        let results = arr.
        <label
          htmlFor="answer"
        >
          METHOD
        </label>
        <input
          id="answer"
          type="text"
        />
        (val =&gt; val &gt; 0)
      </p>
      <p>
        console.log(results) // logs true
      </p>
      <h3>
        PRESS ENTER TO SUBMIT
      </h3>
    </form>
  </section>
</Snippet>
`;
